### PR TITLE
Arrow Format Support

### PR DESCRIPTION
**Pull request by sfc-gh-jbahk**
### Description
Support Arrow by converting first chunk (base64) and the remaining downloaded chunks to Arrow and using native data structures to decode and convert rows to a native Golang datatype.

### Checklist
- [x] Code compiles correctly
- [x] Run ``make fmt`` to fix inconsistent formats
- [x] Run ``make lint`` to get lint errors and fix all of them
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Comment by howryu** [comment]
Can you split the PR into smaller pieces? e.g. maybe check in arrow code under vendor first, then check in handle small result set case etc?

**Comment by sfc-gh-stakeda** [comment]
I'm fine. I'll just ignore all changes under `vendor`.

**Comment by sfc-gh-stakeda** [comment]
Run `make fmt lint` to fix the format and small fixes as well.

File: chunk.go [hunk 1]
```diff
@@ -47,7 +47,10 @@ func decodeLargeChunk(r io.Reader, rowCount int, cellCount int) ([][]*string, er
 		return nil, err
 	}
 
-	return rows, nil
+	chunkRows := make([]chunkRowType, len(rows))
+	populateJSONRowSet(chunkRows, rows)
```
Comment (sfc-gh-tchen) [inline chunk.go:51]: hide the logic of making chunkRows inside populateJSONRowSet and make the signature of populateJSONRowSet as: chunkRows = populateJsonRowSet(rows), this way you have a clear input/output.

**Comment by sfc-gh-stakeda** [comment]
Ok, let's wait for the GS to include the go arrow format flag.

**Pull request merged**